lib/fastlane/actions/sonar.rb

```diff
@@ -0,0 +1,86 @@
+module Fastlane
+  module Actions
+    class SonarAction < Action
+      def self.run(params)
+        verify_sonar_runner_binary
+
+        command_prefix = [
+          'cd',
+          File.expand_path('.').shellescape,
+          '&&'
+        ].join(' ')
+
+        sonar_runner_args = []
+        sonar_runner_args << "-Dproject.settings=\"#{params[:project_configuration_path]}\"" if params[:project_configuration_path]
+        sonar_runner_args << "-Dsonar.projectKey=\"#{params[:project_key]}\"" if params[:project_key]
+        sonar_runner_args << "-Dsonar.projectName=\"#{params[:project_name]}\"" if params[:project_name]
+        sonar_runner_args << "-Dsonar.projectVersion=\"#{params[:project_version]}\"" if params[:project_version]
+        sonar_runner_args << "-Dsonar.sources=\"#{params[:sources_path]}\"" if params[:sources_path]
+
+        command = [
+          command_prefix,
+          'sonar-runner',
+          sonar_runner_args
+        ].join(' ')
+
+        Action.sh command
+      end
+
+      def self.verify_sonar_runner_binary
+        raise "You have to install sonar-runner using `brew install sonar-runner`".red unless `which sonar-runner`.to_s.length > 0 or !Helper.test?
```

Comment:
Please use `||` instead of `or`, as the order is unclear otherwise :+1: Or even better: make it 3 lines instead of one